File: tools/xmc/examples/c-java-unified/bridge.c
#include <stdint.h>
#include <stdio.h>

/*
 * Safe experience prototype: the native side exposes a tiny C ABI.
 * It does not expose C++ object layouts or arbitrary process memory.
 */

int asysma_native_start(uint32_t version) {
    if (version != 1) return -1;
    puts("ASYSMA native layer: started");
    return 0;
}

File: tools/xmc/examples/c-java-unified/Bridge.java
package examples.cjava;

/**
 * Managed half of the C/Java ASYSMA experience prototype.
 * The production bridge should use a documented JNI or FFM boundary.
 */
public final class Bridge {
    private Bridge() {}

    public static void main(String[] args) {
        System.out.println("ASYSMA Java layer: started");
        System.out.println("Native ABI contract: version 1");
    }
}
